.github/workflows/workflow-dispatch-job.yml

```diff
@@ -53,63 +53,125 @@ jobs:
         include: ${{ fromJSON(startsWith(inputs.runner, 'linux') && inputs.dummy_matrix || '[]') }}
     runs-on: ${{inputs.runner}}
     container:
-      options: -u root
-      image: ${{inputs.image}}
+      image: cruizba/ubuntu-dind:jammy-26.1.3
       env:
-        NVIDIA_VISIBLE_DEVICES: ${{ env.NVIDIA_VISIBLE_DEVICES }}
+        NVIDIA_VISIBLE_DEVICES: ${{env.NVIDIA_VISIBLE_DEVICES}}
     steps:
+      - name: Install dependencies
+        shell: sh
+        run: |
+          # Install script dependencies
+          apt update
+          apt install -y --no-install-recommends tree git
       - name: Checkout repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           path: ${{github.event.repository.name}}
           persist-credentials: false
-      - name: Link files to coder user home directory
-        run: |
-          ln -s "$(pwd)/${{github.event.repository.name}}" /home/coder/${{github.event.repository.name}}
-          chown -R coder:coder ${{github.event.repository.name}}
-          chown -R coder:coder /home/coder/${{github.event.repository.name}}
```

Comment:
@alliepiper I believe these lines mean that currently, CI jobs are populating the cache for objects built under `${{github.workspace}}/cccl`, (i.e. `/__w/cccl/cccl/cccl`), because sccache + compilers dereference symlinks in paths to includes and input files.

I've been observing local builds in the devcontainers aren't hitting the cache populated by CI jobs, which makes sense since they build under the absolute path `/home/coder/cccl`.

This PR updates that by mounting the source into the container at `/home/coder/cccl` (L164 in the new version), so CI jobs are always building at this absolute path instead of under the github workspace.